tests/args_names.py

```diff
@@ -0,0 +1,87 @@
+import unittest
+import shlex
+from re import compile
+
+from pykickstart.sections import PackageSection, PreScriptSection, PreInstallScriptSection,\
+    PostScriptSection, OnErrorScriptSection, TracebackScriptSection
+from pykickstart.version import makeVersion
+
+VALID_KICKSTART_OPTION_PATTERN = r"--[a-z0-9\-]+"
+
+
+class ArgumentNamesStatic_TestCase(unittest.TestCase):
+
+    def setUp(self):
+        self._bad_options = []
+        self._reg_name_checker = compile(VALID_KICKSTART_OPTION_PATTERN)
+        self._handler = makeVersion()
+
+    def runTest(self):
+
+        self._check_commands()
+        self._check_sections()
+        self._report()
+
+    def _check_sections(self):
+
+        sections = {
+            "%packages": PackageSection,
+            "%pre": PreScriptSection,
+            "%pre-install": PreInstallScriptSection,
+            "%post": PostScriptSection,
+            "%traceback": TracebackScriptSection,
+            "%onerror": OnErrorScriptSection,
+        }
+
+        for section, secClass in sections.items():
+            instance = secClass(self._handler)
+            arg_parser = instance._getParser()
+
+            self._check_parser_actions(arg_parser, section)
+
+    def _check_commands(self):
+
+        for command, cmdClass in self._handler.commandMap.items():
+
+            if command == "method":
+                continue
+
+            args = shlex.split(command, comments=True)
+            cmd = args[0]
+
+            ks_parser = self._handler.commands[cmd]
+            ks_parser.currentLine = command
+            ks_parser.currentCmd = args[0]
+            ks_parser.seen = True
+            arg_parser = ks_parser._getParser()
+
+            self._check_parser_actions(arg_parser, command, cmd_class=cmdClass)
+
+    def _check_parser_actions(self, arg_parser, cmd_name, cmd_class=None):
+
+        for action in arg_parser._get_optional_actions():
+            if action.deprecated:
+                continue
+
+            found_any_good = False
+            for option in action.option_strings:
+                if cmd_class and option.lstrip("-") in cmd_class.removedAttrs:
+                    # caution removedAttrs does not include leading dashes
+                    continue
+                is_good = self._reg_name_checker.fullmatch(option) is not None
+                found_any_good = found_any_good or is_good
+                if not is_good:
+                    print("Found option with uppercase letters: {} for command {}.".format(
```

Comment:
Let's try not to use .format() unless it really adds something that '%' doesn't do. In general pykickstart has only used %s and %(dictname)s in the string formatting so let's stick with that.
Also, for long lines, I like to err on the side of readability vs. wrapping at someone else's edit width so I don't think this needs to be on 2 lines.